homeassistant/components/notion/__init__.py

```diff
@@ -180,6 +180,10 @@ def async_save_refresh_token(refresh_token: str) -> None:
     # Create a callback to save the refresh token when it changes:
     entry.async_on_unload(client.add_refresh_token_callback(async_save_refresh_token))
 
+    # Save our current refresh token:
+    if client.refresh_token:
+        async_save_refresh_token(client.refresh_token)
```

Comment:
This means that you will always update the entry. Should you verify that it has changed before saving it ?